### PR TITLE
"Extra strong" Lucas pseudoprime

### LINKED ISSUE #18921: ntheory/primetest/is_extra_strong_lucas_prp edge case
In the cited paper "Frobenius Pseudoprimes" p. 876, the second condition on being an "extra strong" Lucas pseudoprime reads:

    V_{2^ts} ≡ 0 mod n, where 0 ≤ t < r-1

However, the code for [is_extra_strong_lucas_prp()](https://github.com/sympy/sympy/blob/126485017c60e249c21dd68c46f70ea7b0584e1e/sympy/ntheory/primetest.py#L401) reads:

    if V == 0:
        return True
    for r in range(1, s):
        V = (V*V - 2) % n
        if V == 0:
            return True

When `V == 0` and `s == 1`, by the above definition ("s" is "r" there), the number should not be declared as a pseudoprime. Perhaps the code should read as:

    for r in range(1, s):
        if V == 0:
            return True
        V = (V*V - 2) % n

### PR DESCRIPTION
**Pull request by Arpan612**
"Extra strong" Lucas pseudoprime 2nd condition corrected
#### References to other Issues or PRs

Fixes #18921 

#### Brief description of what is fixed or changed
In the cited paper "Frobenius Pseudoprimes" p. 876, the second condition on being an "extra strong" Lucas pseudoprime reads:

V_{2^ts} ≡ 0 mod n, where 0 ≤ t < r-1

When V == 0 and s == 1, by the above definition ("s" is "r" there), the number should not be declared as a pseudoprime. Hence the change is needed.

#### Release Notes

* ntheory
   * Corrects the condition for "Extra strong" Lucas pseudoprime

### CODE REVIEW

**Comment by Arpan612** [comment]
@sylee957 Sir, could you please review the PR.

**Comment by sylee957** [comment]
I think that there is no way to make this change visible.
If `s` is 1, `V` seems like always not divisible by `n`. But I haven't got through how to prove this.

**Comment by Arpan612** [comment]
@sylee957 Sir, do you expect a change in PR then. Or at least could you share some documents for me to explore on this issue?

**Comment by Arpan612** [comment]
@sukop Could you help me here?

> I think that there is no way to make this change visible.
> If `s` is 1, `V` seems like always not divisible by `n`. But I haven't got through how to prove this.

**Comment by sylee957** [comment]
I think that you should squash the commits because there are more redundant commits than the code itself.

My comment above is about the difficulty to getting the counterexample that is fixed.

**Comment by czgdp1807** [comment]
I have cleared the commits. Just make some minor changes so that your name is recorded as well in the git history.

**Comment by czgdp1807** [comment]
Is there any test case that can be added for this fix?

**Comment by Arpan612** [comment]
> Is there any test case that can be added for this fix?

Sir, actually I am working on two other issues right now. Is it okay if you merge this PR for now, and as soon as I have a test case update, I will send another PR? Actually, GSoC has the mandatory merged PR rule before 31st, that is why.

**Comment by Arpan612** [comment]
@czgdp1807 Also sir, could you please tell me how did you manage the squash commit? My attempt was continuously showing compatibility issues.

**Comment by Arpan612** [comment]
@czgdp1807 Sir, I have added the test cases.

File: sympy/ntheory/primetest.py [hunk 1]
```diff
@@ -461,15 +468,13 @@ def is_extra_strong_lucas_prp(n):
     k = (n+1) >> s
 
     U, V, Qk = _lucas_sequence(n, P, Q, k)
-
+#changes made
```
Comment (mohitacecode) [inline sympy/ntheory/primetest.py]: Not needed

File: sympy/ntheory/primetest.py [hunk 2]
```diff
@@ -443,6 +443,13 @@ def is_extra_strong_lucas_prp(n):
     #   2) The MathWorld page as of June 2013 specifies Q=-1.  The Lucas
     #      sequence must have Q=1.  See Grantham theorem 2.3, any of the
     #      references on the MathWorld page, or run it and see Q=-1 is wrong.
+
+    """
+    Test cases:
+    V = 0, s= 1, not a pseudoprime
+    V = 0, s= 3, pseudoprime
+    V = 0, s= 5, pseudoprime
+    """
```
Comment (sylee957) [inline sympy/ntheory/primetest.py]: I think that this docstring can just error out or ignored, so it should be removed.

**Comment by mohitacecode** [comment]
you have to add test in `test_primetest.py` file.

**Comment by mohitacecode** [comment]
This is not how you should write test take a look at `test_primetest.py` you will get it.

File: sympy/ntheory/primetest.py (hunk 2 above)
Comment (Arpan612) [reply sympy/ntheory/primetest.py]: Okay sir

**Comment by Arpan612** [comment]
@mohitacecode Got it. Working on it.

**Comment by Arpan612** [comment]
@sylee957 @czgdp1807 Could you please review the changes?

**Comment by Arpan612** [comment]
@sylee957 Thank you for the correction.

**Pull request merged**